:bomb: `Bug`: Apply failed => some error(s) occurred during plugins apply process

### What Happened?

./dtm apply -y
```
2022-12-04 23:12:04 ℹ [INFO]  Apply started.
2022-12-04 23:12:05 ℹ [INFO]  Using local backend. State file: devstream.state.
2022-12-04 23:12:05 ℹ [INFO]  Tool (repo-scaffolding/golang-github) found in config but doesn't exist in the state, will be created.
2022-12-04 23:12:05 ℹ [INFO]  Tool (githubactions-golang/default) found in config but doesn't exist in the state, will be created.
2022-12-04 23:12:05 ℹ [INFO]  Start executing the plan.
2022-12-04 23:12:05 ℹ [INFO]  Changes count: 2.
2022-12-04 23:12:05 ℹ [INFO]  -------------------- [  Processing progress: 1/2.  ] --------------------
2022-12-04 23:12:05 ℹ [INFO]  Processing: (repo-scaffolding/golang-github) -> Create ...
2022-12-04 23:12:05 ℹ [INFO]  github start to download repoTemplate...
2022-12-04 23:12:07 ⚠ [WARN]  repo render file failed: template: go-webapp-devstream-demo:19:18: executing "go-webapp-devstream-demo" at <.imageRepo>: map has no entry for key "imageRepo"
2022-12-04 23:12:07 ⚠ [WARN]  repoScaffolding render repoTemplate failed=> template: go-webapp-devstream-demo:19:18: executing "go-webapp-devstream-demo" at <.imageRepo>: map has no entry for key "imageRepo"
2022-12-04 23:12:07 !! [ERROR]  repo-scaffolding/golang-github Create failed with error: render RepoTemplate files failed
2022-12-04 23:12:07 ℹ [INFO]  -------------------- [  Processing aborted.  ] --------------------
2022-12-04 23:12:07 !! [ERROR]  Errors Map: key(repo-scaffolding/golang-github/Create) -> value(render RepoTemplate files failed)
2022-12-04 23:12:07 !! [ERROR]  Apply failed => some error(s) occurred during plugins apply process.
```


### How to Reproduce?

./dtm show config -t quickstart > config.yaml
sed -i "s@YOUR_GITHUB_USERNAME_CASE_SENSITIVE@${GITHUB_USER}@g" config.yaml
sed -i "s@YOUR_DOCKER_USERNAME@${DOCKERHUB_USERNAME}@g" config.yaml
./dtm init
./dtm apply -y

### Anything else

DevStream Version 0.10.1
Linux manjaro-xfce-21.3.7-220816-linux515

### DevStream Version

latest

--- a/docs/core-concepts/output.md
+++ b/docs/core-concepts/output.md
@@ -57,7 +57,7 @@ tools:
       branch: main
       repoType: github
     vars:
-      ImageRepo: "ironcore864/golang-demo"
+      imageRepo: "ironcore864/golang-demo"
     sourceRepo:
       org: devstream-io
       repo: dtm-scaffolding-golang

--- a/docs/plugins/argocdapp.md
+++ b/docs/plugins/argocdapp.md
@@ -45,7 +45,7 @@ tools:
       branch: [[ defaultBranch ]]
       repoType: github
     vars:
-      ImageRepo: "[[ dockerhubUsername ]]/[[ repoName ]]"
+      imageRepo: "[[ dockerhubUsername ]]/[[ repoName ]]"
     sourceRepo:
       org: devstream-io
       repo: dtm-scaffolding-golang

--- a/docs/plugins/repo-scaffolding.md
+++ b/docs/plugins/repo-scaffolding.md
@@ -116,7 +116,7 @@ tools:
       repo: dtm-scaffolding-golang
       repoType: github
     vars:
-      ImageRepo: dtm-test/golang-repo
+      imageRepo: dtm-test/golang-repo
 ```
 
 This config will create `dtm-test-golang` repo for user test_owner in GitHub, and the variable ImageRepo will be used for template render. 

--- a/examples/gitops.yaml
+++ b/examples/gitops.yaml
@@ -30,7 +30,7 @@ tools:
       repo: dtm-scaffolding-golang
       repoType: github
     vars:
-      ImageRepo: "[[ dockerhubUsername ]]/[[ repoName ]]"
+      imageRepo: "[[ dockerhubUsername ]]/[[ repoName ]]"
 - name: jira-github-integ
   instanceID: default
   dependsOn: [ "repo-scaffolding.golang-github" ]

--- a/examples/quickstart.yaml
+++ b/examples/quickstart.yaml
@@ -19,7 +19,7 @@ tools:
       repo: dtm-scaffolding-golang
       repoType: github
     vars:
-      ImageRepo: YOUR_DOCKER_USERNAME/go-webapp-devstream-demo
+      imageRepo: YOUR_DOCKER_USERNAME/go-webapp-devstream-demo
 - name: githubactions-golang
   instanceID: default
   dependsOn: ["repo-scaffolding.golang-github"]

--- a/internal/pkg/plugin/installer/reposcaffolding/installer.go
+++ b/internal/pkg/plugin/installer/reposcaffolding/installer.go
@@ -35,8 +35,7 @@ func InstallRepo(options configmanager.RawOptions) error {
 		processRepoFileFunc(appName, opts.renderTplConfig()),
 	)
 	if err != nil {
-		log.Warnf("repoScaffolding render repoTemplate failed=> %+v", err)
-		return fmt.Errorf("render RepoTemplate files failed")
+		return fmt.Errorf("render RepoTemplate files failed with error: %w", err)
 	}
 
 	// 3. push repo to DestinationRepo

--- a/internal/pkg/show/config/plugins/repo-scaffolding.yaml
+++ b/internal/pkg/show/config/plugins/repo-scaffolding.yaml
@@ -23,4 +23,4 @@ tools:
       repoType: YOUR_TEMPLATE_REPO_TYPE
     # this is used for template render
     vars:
-      ImageRepo: YOUR_DOCKERHUB_REPOSITORY
+      imageRepo: YOUR_DOCKERHUB_REPOSITORY

--- a/internal/pkg/show/config/templates/gitops.yaml
+++ b/internal/pkg/show/config/templates/gitops.yaml
@@ -30,7 +30,7 @@ tools:
       repo: dtm-scaffolding-golang
       repoType: github
     vars:
-      ImageRepo: "[[ dockerhubUsername ]]/[[ repoName ]]"
+      imageRepo: "[[ dockerhubUsername ]]/[[ repoName ]]"
 - name: jira-github-integ
   instanceID: default
   dependsOn: [ "repo-scaffolding.golang-github" ]

--- a/internal/pkg/show/config/templates/quickstart.yaml
+++ b/internal/pkg/show/config/templates/quickstart.yaml
@@ -1,5 +1,5 @@
 config:
-  state:
+  state: # state config, backend can be local, s3 or k8s
     backend: local
     options:
       stateFile: devstream.state

--- a/test/e2e/yaml/e2e-test-local.yaml
+++ b/test/e2e/yaml/e2e-test-local.yaml
@@ -23,7 +23,7 @@ tools:
       branch: [[ defaultBranch ]]
       repoType: github
     vars:
-      ImageRepo: "[[ dockerhubUsername ]]/[[ repoName ]]"
+      imageRepo: "[[ dockerhubUsername ]]/[[ repoName ]]"
     sourceRepo:
       org: devstream-io
       repo: dtm-scaffolding-golang

